app/code/Magento/Swatches/Helper/Data.php

```diff
@@ -353,7 +353,7 @@ private function getAllSizeImages(ModelProduct $product, $imageFile)
             'large' => $this->imageHelper->init($product, 'product_page_image_large_no_frame')
                 ->setImageFile($imageFile)
                 ->getUrl(),
-            'medium' => $this->imageHelper->init($product, 'product_page_image_medium_no_frame')
+            'medium' => $this->imageHelper->init($product, 'new_products_content_widget_grid')
```

Comment:
I'm not sure `new_products_content_widget_grid` is a correct identifier in this case. We have `product_page_image_*` for every type of image and that looks logical. The `new_products_content_widget_grid` might work, however, this identifier belongs to the different area (new products widget). If we introduce such change we are not sure our swatches will work for product listing and product details pages after some changes in the new products widget. We need to figure out why the image identifier from the same area (`product_page_image_*` ) does not work here. 
Check the implementation in 2.3-develop branch, maybe it will get some clue how we can handle this situation.